Sources/SwiftParser/Directives.swift

```diff
@@ -121,14 +124,23 @@ extension Parser {
           } else {
             condition = nil
           }
+        case .pound:
```

Comment:
Your implementation currently also constructs a `#elseif` for everything other than `#elif`. For example if you have

```swift
#if MY_FLAG
#line
#endif
```

I think this would treat `#line` just like `#elseif`, producing a diagnostic, even though the code is valid.

Also, we need to consider that someone might have declared a macro named `elif` that doesn’t take any parameters and thus even

```swift
#if MY_FLAG
#elif
#endif
```

might be valid Swift (it’s not yet because [SE-0382](https://github.com/apple/swift-evolution/blob/main/proposals/0382-expression-macros.md) hasn’t been accepted yet, but it will probably be valid soon).

So, what I think you need to do is, first thing in the loop, even before `self.eat(poundIfHandle)`, you need to check that 
(a) the `#` is followed by `elif` and 
(b) that `elif` is followed by another identifier which is on the same line as `elif`. 

(b) should ensure that we can disambiguate between the macro case and `#elif` which was meant to be `#elseif`

Reply:
Thank you for the tip. I'm trying to accomplish what you suggested. I'll submit the code once it's ready.

Reply:
@ahoppen I'm doing the check in the following code:
```swift
         var atElifTypo: Bool {
           guard self.at(.pound) else {
             return false
           }

           var lookahead = self.lookahead()
           lookahead.consumeAnyToken()

           if lookahead.at(.identifier), lookahead.currentToken.tokenText == "elif" {
             lookahead.consumeAnyToken()
           } else {
             return false
           }

           return lookahead.at(.identifier)
         }
```

The tests are passing, although I'm not sure if this is enough.